src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/StartAppealMidEvent.java

```diff
@@ -53,6 +55,12 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
 
         PreSubmitCallbackResponse<AsylumCase> response = new PreSubmitCallbackResponse<>(asylumCase);
 
+        if (callback.getPageId().equals(DETENTION_FACILITY_PAGE_ID)) {
```

Comment:
here on line 58, please check event as well. As for EditAppealAfterSubmit event and (even in future if we have) pageId DetentionFacility, do we need the code to run?

Reply:
Added event check along with pageId check